packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx

```diff
@@ -1,66 +1,75 @@
-    test(`${variant} helper text variant applies successfully`, () => {
-      const { asFragment } = render(
-        <HelperTextItem variant={variant as 'default' | 'indeterminate' | 'warning' | 'success' | 'error'}>
-          {variant} help test text
-        </HelperTextItem>
-      );
-      expect(asFragment()).toMatchSnapshot();
-    });
-  });
+test('Renders to match snapshot', () => {
+  const { asFragment } = render(
+    <HelperText>
+      <HelperTextItem>help test text</HelperTextItem>
+      <HelperTextItem>help test text 2</HelperTextItem>
+      <HelperTextItem>help test text 3</HelperTextItem>
+    </HelperText>
+  );
+  expect(asFragment()).toMatchSnapshot();
+});
 
-  test('variant comonent helper text renders properly', () => {
-    const { asFragment } = render(
-      <HelperText component="ul">
-        <HelperTextItem component="li">help test text 1</HelperTextItem>
-        <HelperTextItem component="li">help test text 2</HelperTextItem>
-      </HelperText>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('HelperText renders custom className', () => {
+  render(
+    <HelperText className="custom" data-testid="helper">
+      <HelperTextItem>help test text 1</HelperTextItem>
+    </HelperText>
+  );
+  expect(screen.getByTestId('helper')).toHaveClass('custom');
+});
 
-  test('icon helper text renders properly', () => {
-    const { asFragment } = render(
-      <HelperText>
-        <HelperTextItem icon={<CheckIcon />}>help test text</HelperTextItem>
-      </HelperText>
+Object.values(['indeterminate', 'warning', 'success', 'error']).forEach((variant) => {
+  test(`Renders ${styles.modifiers[variant]} when variant = ${variant}`, () => {
+    render(
+      <HelperTextItem variant={variant as 'default' | 'indeterminate' | 'warning' | 'success' | 'error'}>
+        text
+      </HelperTextItem>
     );
-    expect(asFragment()).toMatchSnapshot();
+    expect(screen.getByText('text').parentElement).toHaveClass(styles.modifiers[variant]);
   });
+});
 
-  test('dynamic helper text renders successfully', () => {
-    const { asFragment } = render(
-      <HelperText>
-        <HelperTextItem isDynamic>help test text</HelperTextItem>
-      </HelperText>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('HelperTextItem renders custom className', () => {
+  render(
+    <HelperText>
+      <HelperTextItem className="custom">help test text 1</HelperTextItem>
+    </HelperText>
+  );
+  expect(screen.getByText('help test text 1').parentElement).toHaveClass('custom');
+});
 
-  test('helper text block renders successfully', () => {
-    const { asFragment } = render(
-      <HelperText>
-        <HelperTextItem>help test text 1</HelperTextItem>
-        <HelperTextItem>help test text 2</HelperTextItem>
-        <HelperTextItem>help test text 3</HelperTextItem>
-      </HelperText>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders as ul when component = ul', () => {
```

Comment:
```suggestion
test('Renders with element passed to component prop', () => {
```

We should also have a test that checks the top element is a div by default